core/src/Revolution/Processors/Workspace/PackageNamespace/GetList.php

```diff
@@ -55,33 +62,94 @@ public function initialize()
-                'OR:path:LIKE' => '%' . $search . '%',
+                'name:LIKE' => '%' . $query . '%',
+                'OR:path:LIKE' => '%' . $query . '%',
             ]);
         }
+
+        // $foreignKey is the primary key of the child settings entity (e.g., user, usergroup, context)
+        $foreignKey = $this->getProperty('foreignKey', '');
+        $foreignKeyWhere = null;
+
         /*
             When this class is used to fetch data for a grid filter's store (combo),
             limit results to only those namespaces present in the current grid.
         */
-        if ($this->isGridFilter) {
-            if ($userGroup = $this->getProperty('usergroup', false)) {
-                $c->innerJoin(
-                    modAccessNamespace::class,
-                    'modAccessNamespace',
-                    [
-                        '`modAccessNamespace`.`target` = `modNamespace`.`name`',
-                        '`modAccessNamespace`.`principal` = ' . (int)$userGroup,
-                        '`modAccessNamespace`.`principal_class` = ' . $this->modx->quote(modUserGroup::class)
-                    ]
-                );
-                if ($policy = $this->getProperty('policy', false)) {
-                    $c->where([
-                        '`modAccessNamespace`.`policy`' => (int)$policy
+        if ($this->isGridFilter && $targetGrid = $this->getProperty('targetGrid', false)) {
+            switch($targetGrid) {
+                case 'MODx.grid.SettingsGrid':
+                    $settingsType = $this->getProperty('targetSettingsType', 'system');
+                    $alias = 'settingsNamespace';
+                    switch($settingsType) {
+                        case 'context':
+                            $settingsClass = modContextSetting::class;
+                            $foreignKeyWhere = $foreignKey ? [ $alias . '.context_key' => $this->modx->sanitizeString($foreignKey) ] : null ;
+                            break;
+                        case 'group':
+                            $settingsClass = modUserGroupSetting::class;
+                            $foreignKeyWhere = $foreignKey ? [ $alias . '.group' => (int)$foreignKey ] : null ;
+                            break;
+                        case 'system':
+                            $settingsClass = modSystemSetting::class;
+                            break;
+                        case 'user':
+                            $settingsClass = modUserSetting::class;
+                            $foreignKeyWhere = $foreignKey ? [ $alias . '.user' => (int)$foreignKey ] : null ;
+                            break;
+                        // no default
+                    }
+
+                    $nsSubquery = $this->modx->newQuery($settingsClass);
+                    $nsSubquery->setClassAlias($alias);
+                    $nsSubquery->select([
+                        'namespaces' => "GROUP_CONCAT(DISTINCT `{$alias}`.`namespace` SEPARATOR '\",\"')"
```

Comment:
Is `SEPARATOR` something special in queries?

Reply:
Yes, in the CONCAT aggregate functions the `SEPARATOR` is inserted between the values found. The default separator is just a comma, but I need the double quotes as well since the concatenated list being built consists of strings and not integers, and the found list is ultimately being used for the `where IN` condition later on.